source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -780,10 +799,21 @@ void InjaTransformer::transform(Http::RequestOrResponseHeaderMap &header_map,
   for (const auto &templated_dynamic_metadata : dynamic_metadata_) {
     std::string output = instance_->render(templated_dynamic_metadata.template_);
     if (!output.empty()) {
-      ProtobufWkt::Struct strct(
-          MessageUtil::keyValueStruct(templated_dynamic_metadata.key_, output));
-      callbacks.streamInfo().setDynamicMetadata(
-          templated_dynamic_metadata.namespace_, strct);
+      // Need to check if number
+      try {
+        const auto int_output = std::stoi(output);
+        ProtobufWkt::Struct struct_obj;
+        ProtobufWkt::Value val;
+        val.set_number_value(int_output);
+        (*struct_obj.mutable_fields())[templated_dynamic_metadata.key_] = val;
+        callbacks.streamInfo().setDynamicMetadata(
+            templated_dynamic_metadata.namespace_, struct_obj);
+      } catch (...) { // This function can return multiple exceptions
```

Comment:
what here can throw an exception? is there a way to avoid it?
the problem with `catch(...)` is that we have no idea what went wrong.. this will be hard to debug.

Reply:
stoi can throw 2 exceptions, neither are super helpful here.

https://en.cppreference.com/w/cpp/string/basic_string/stol

Reply:
I can catch them if you want for better logging

Reply:
I got rid of this completely